diff --git a/ppapi/buildflags/buildflags.gni b/ppapi/buildflags/buildflags.gni
index 0bc0e153dff7e..fba1a83be000e 100644
--- a/ppapi/buildflags/buildflags.gni
+++ b/ppapi/buildflags/buildflags.gni
@@ -19,7 +19,7 @@ declare_args() {
   # Enables Pepper API (PPAPI) plugin support.
   # The only remaining PPAPI plugin supported is NaCl and if is disabled
   # we can disable PPAPI.
-  enable_ppapi = enable_plugins && enable_nacl && !is_fuchsia
+  enable_ppapi = enable_plugins && !is_fuchsia
 }
 
 assert(enable_plugins || !enable_ppapi)
