diff --git a/ui/base/ime/init/input_method_initializer.cc b/ui/base/ime/init/input_method_initializer.cc
index bddb5c062a89f..a500669ac54e7 100644
--- a/ui/base/ime/init/input_method_initializer.cc
+++ b/ui/base/ime/init/input_method_initializer.cc
@@ -32,16 +32,6 @@ void ShutdownInputMethod() {
 #endif
 }
 
-void RestartInputMethod() {
-#if !BUILDFLAG(IS_CHROMEOS_ASH) && BUILDFLAG(IS_WIN)
-  // Some tests don't have a TSFBridge and may cause error if we set one.
-  if (TSFBridge::GetInstance()) {
-    TSFBridge::Shutdown();
-    TSFBridge::Initialize();
-  }
-#endif
-}
-
 void InitializeInputMethodForTesting() {
 #if defined(USE_AURA) && BUILDFLAG(IS_LINUX)
   GetInputMethodContextFactoryForTest() =
