diff --git a/third_party/.gitignore b/third_party/.gitignore
index 1a18347e45e1e..b0007c4b305dc 100644
--- a/third_party/.gitignore
+++ b/third_party/.gitignore
@@ -51,7 +51,9 @@
 /cygwin
 /devserver
 /directxsdk
+/electron_node
 /elfutils/src
+/engflow-reclient-configs/
 /espresso/lib/
 /eyesfree/src
 /fuchsia-sdk/images
@@ -104,6 +106,7 @@
 /mocha
 /mockito/src
 /nacl_sdk_binaries/
+/nan
 /ninja/ninja*
 /node/*.tar.gz
 /node/linux/
@@ -149,6 +152,7 @@
 /soda-win64
 /speex
 /sqlite4java/lib/
+/squirrel.mac
 /subresource-filter-ruleset/data/UnindexedRules
 /swift-format
 /swift-toolchain
