diff --git a/chrome/browser/prefs/BUILD.gn b/chrome/browser/prefs/BUILD.gn
index 60e08d69c11f1..60abe2e076563 100644
--- a/chrome/browser/prefs/BUILD.gn
+++ b/chrome/browser/prefs/BUILD.gn
@@ -6,6 +6,7 @@ import("//components/enterprise/buildflags/buildflags.gni")
 import("//components/safe_browsing/buildflags.gni")
 import("//extensions/buildflags/buildflags.gni")
 import("//pdf/features.gni")
+import("//third_party/ipfs_client/args.gni")
 
 assert(is_win || is_mac || is_linux || is_chromeos || is_android)
 
@@ -124,7 +125,9 @@ source_set("impl") {
     "//services/preferences/public/mojom",
     "//services/preferences/tracked",
   ]
-
+  if (enable_ipfs) {
+    deps += [ "//components/ipfs" ]
+  }
   if (is_chromeos) {
     deps += [
       "//ash",
