diff --git a/build/config/linux/pkg-config.py b/build/config/linux/pkg-config.py
index f4d31de6769f3..54f9361121744 100755
--- a/build/config/linux/pkg-config.py
+++ b/build/config/linux/pkg-config.py
@@ -224,6 +224,7 @@ def main():
       lib_dirs.append(RewritePath(flag[2:], prefix, sysroot))
     elif flag[:2] == '-I':
       includes.append(RewritePath(flag[2:], prefix, sysroot))
+      includes.append(flag[2:])
     elif flag[:3] == '-Wl':
       # Don't allow libraries to control ld flags.  These should be specified
       # only in build files.
