diff --git a/chrome/browser/BUILD.gn b/chrome/browser/BUILD.gn
index a188528a9e262..88df13b162858 100644
--- a/chrome/browser/BUILD.gn
+++ b/chrome/browser/BUILD.gn
@@ -40,6 +40,7 @@ import("//rlz/buildflags/buildflags.gni")
 import("//sandbox/features.gni")
 import("//testing/libfuzzer/fuzzer_test.gni")
 import("//third_party/blink/public/public_features.gni")
+import("//third_party/ipfs_client/args.gni")
 import("//third_party/protobuf/proto_library.gni")
 import("//third_party/webrtc/webrtc.gni")
 import("//third_party/widevine/cdm/widevine.gni")
@@ -1912,7 +1913,6 @@ static_library("browser") {
       "user_education/user_education_service_factory.h",
     ]
   }
-
   configs += [
     "//build/config/compiler:wexit_time_destructors",
     "//build/config:precompiled_headers",
@@ -2604,6 +2604,14 @@ static_library("browser") {
       ]
     }
 
+    if (enable_ipfs) {
+      sources += [
+        "ipfs_extra_parts.cc",
+        "ipfs_extra_parts.h",
+      ]
+      deps += [ "//components/ipfs" ]
+    }
+
     if (is_chromeos_ash) {
       deps += [ "//chrome/browser/screen_ai:screen_ai_dlc_installer" ]
     }
