diff --git a/components/cbor/reader.h b/components/cbor/reader.h
index f0b43a5517528..a57e277a1bc66 100644
--- a/components/cbor/reader.h
+++ b/components/cbor/reader.h
@@ -130,6 +130,11 @@ class CBOR_EXPORT Reader {
     // during decoding will set raise the `UNSUPPORTED_FLOATING_POINT_VALUE`
     // error.
     bool allow_floating_point = false;
+
+    // If the parser encounters a TAG element, should it be parsed out and
+    // the tag value saved (true), or should the entire node and its content
+    // be discarded (false)
+    bool parse_tags = false;
   };
 
   Reader(const Reader&) = delete;
@@ -204,6 +209,9 @@ class CBOR_EXPORT Reader {
   absl::optional<Value> ReadMapContent(const DataItemHeader& header,
                                        const Config& config,
                                        int max_nesting_level);
+  absl::optional<Value> ReadTagContent(const DataItemHeader& header,
+                                       const Config& config,
+                                       int max_nesting_level);
   absl::optional<uint8_t> ReadByte();
   absl::optional<base::span<const uint8_t>> ReadBytes(uint64_t num_bytes);
   bool IsKeyInOrder(const Value& new_key,
