diff --git a/components/enterprise/buildflags/buildflags.gni b/components/enterprise/buildflags/buildflags.gni
index fa27ba6cfe55f..1735969ab7b12 100644
--- a/components/enterprise/buildflags/buildflags.gni
+++ b/components/enterprise/buildflags/buildflags.gni
@@ -41,4 +41,4 @@ declare_args() {
 # These two flags are equivalent, but `enable_print_content_analysis` still
 # needs to exist separately as it lives under //printing which cannot depend
 # on //components.
-assert(enterprise_content_analysis == enable_print_content_analysis)
+# assert(enterprise_content_analysis == enable_print_content_analysis)
