diff --git a/url/url_util.cc b/url/url_util.cc
index ce5225e121f5d..e2ddfd9323214 100644
--- a/url/url_util.cc
+++ b/url/url_util.cc
@@ -308,7 +308,12 @@ bool DoCanonicalize(const CHAR* spec,
     success = CanonicalizeFileSystemURL(
         spec, ParseFileSystemURL(std::basic_string_view(spec, spec_len)),
         charset_converter, output, output_parsed);
-
+  } else if (DoCompareSchemeComponent(spec, scheme, "ipfs")) {
+    auto spec_view = std::basic_string_view(spec, spec_len);
+    // Switch multibase away from case-sensitive ones before continuing canonicalization.
+    auto parsed_input = ParseStandardURL(spec_view);
+    success = CanonicalizeIpfsURL(spec_view, parsed_input, scheme_type,
+                                  charset_converter, output, output_parsed);
   } else if (DoIsStandard(spec, scheme, &scheme_type)) {
     // All "normal" URLs.
     success = CanonicalizeStandardURL(
