diff --git a/net/dns/dns_config_service_linux.cc b/net/dns/dns_config_service_linux.cc
index 253f439f33ae7..9cd457279bab4 100644
--- a/net/dns/dns_config_service_linux.cc
+++ b/net/dns/dns_config_service_linux.cc
@@ -250,20 +250,6 @@ bool IsNsswitchConfigCompatible(
         }
         break;
 
-      case NsswitchReader::Service::kResolve:
-        // If /etc/resolv.conf points to systemd-resolved then treat nss-resolve
-        // the same as nss-dns. If it's not then consider the nsswitch
-        // configuration incompatible.
-        if (!resolv_reader.IsLikelySystemdResolved()) {
-          RecordIncompatibleNsswitchReason(
-              IncompatibleNsswitchReason::kIncompatibleService,
-              specification.service);
-          return false;
-        }
-        // systemd-resolved also supports looking up records from /etc/hosts.
-        files_found = true;
-        [[fallthrough]];
-
       case NsswitchReader::Service::kDns:
         if (!files_found) {
           RecordIncompatibleNsswitchReason(
@@ -287,7 +273,8 @@ bool IsNsswitchConfigCompatible(
         // Ignore any entries after `kDns` because Chrome will fallback to the
         // system resolver if a result was not found in DNS.
         return true;
-
+      case NsswitchReader::Service::kResolve:
+        break;
       case NsswitchReader::Service::kMdns:
       case NsswitchReader::Service::kMdns4:
       case NsswitchReader::Service::kMdns6:
