diff --git a/chrome/browser/profiles/profile_selections.cc b/chrome/browser/profiles/profile_selections.cc
index f363f4a01e221..923fe321a8cd0 100644
--- a/chrome/browser/profiles/profile_selections.cc
+++ b/chrome/browser/profiles/profile_selections.cc
@@ -13,6 +13,7 @@
 #endif  // BUILDFLAG(IS_CHROMEOS_ASH)
 
 bool AreKeyedServicesDisabledForProfileByDefault(const Profile* profile) {
+#if 0
   // By default disable all services for System Profile.
   // Even though having no services is also the default value for Guest Profile,
   // this is not really the case in practice because a lot of Service Factories
@@ -20,7 +21,7 @@ bool AreKeyedServicesDisabledForProfileByDefault(const Profile* profile) {
   if (profile && profile->IsSystemProfile()) {
     return true;
   }
-
+#endif
   return false;
 }
 
@@ -120,6 +121,7 @@ Profile* ProfileSelections::ApplyProfileSelection(Profile* profile) const {
 
 ProfileSelection ProfileSelections::GetProfileSelection(
     Profile* profile) const {
+#if 0
 #if BUILDFLAG(IS_CHROMEOS_ASH)
   // This check has to be performed before the check on
   // `profile->IsRegularProfile()` because profiles that are internal ASH
@@ -155,6 +157,7 @@ ProfileSelection ProfileSelections::GetProfileSelection(
   }
 
   NOTREACHED();
+#endif
   return ProfileSelection::kNone;
 }
 
