diff --git a/chrome/browser/BUILD.gn b/chrome/browser/BUILD.gn
index c88784479b8d6..ab71e2b5da4ee 100644
--- a/chrome/browser/BUILD.gn
+++ b/chrome/browser/BUILD.gn
@@ -38,6 +38,7 @@ import("//sandbox/features.gni")
 import("//services/screen_ai/buildflags/features.gni")
 import("//testing/libfuzzer/fuzzer_test.gni")
 import("//third_party/blink/public/public_features.gni")
+import("//third_party/ipfs_client/args.gni")
 import("//third_party/protobuf/proto_library.gni")
 import("//third_party/webrtc/webrtc.gni")
 import("//third_party/widevine/cdm/widevine.gni")
@@ -2644,6 +2645,13 @@ static_library("browser") {
     # componentized.
     "//chrome/browser/ui/webauthn:impl",
   ]
+  if (enable_ipfs) {
+    sources += [
+      "ipfs_extra_parts.cc",
+      "ipfs_extra_parts.h",
+    ]
+    deps += [ "//components/ipfs" ]
+  }
 
   if (!is_android) {
     deps += [ "//chrome/browser/ui/webui/omnibox_popup/mojom" ]
