diff --git a/chrome/browser/flag_descriptions.h b/chrome/browser/flag_descriptions.h
index fececbd152a27..50e6f7cff8f66 100644
--- a/chrome/browser/flag_descriptions.h
+++ b/chrome/browser/flag_descriptions.h
@@ -23,6 +23,7 @@
 #include "pdf/buildflags.h"
 #include "printing/buildflags/buildflags.h"
 #include "third_party/blink/public/common/buildflags.h"
+#include "third_party/ipfs_client/ipfs_buildflags.h"
 
 // This file declares strings used in chrome://flags. These messages are not
 // translated, because instead of end-users they target Chromium developers and
@@ -537,6 +538,11 @@ inline constexpr char kEnableExtensionsForCorpDesktopAndroidDescription[] =
     "Enables extensions for corporate (google.com and managedchrome.com) "
     "desktop Android dogfooders. Does nothing for other domains.";
 
+#if BUILDFLAG(ENABLE_IPFS)
+inline constexpr char kEnableIpfsName[] = "Enable IPFS";
+inline constexpr char kEnableIpfsDescription[] = "Enable ipfs:// and ipns:// URLs";
+#endif
+
 inline constexpr char kEnableSeamlessSigninName[] = "Enable Seamless Sign-in";
 inline constexpr char kEnableSeamlessSigninDescription[] =
     "Enables the Seamless Sign-in flow that signs in the user without showing "
