diff --git a/url/url_util.cc b/url/url_util.cc
index 0aca7cc1565e2..d34bc8fba9283 100644
--- a/url/url_util.cc
+++ b/url/url_util.cc
@@ -72,6 +72,8 @@ struct SchemeRegistry {
   std::vector<SchemeWithType> referrer_schemes = {
       {kHttpsScheme, SCHEME_WITH_HOST_PORT_AND_USER_INFORMATION},
       {kHttpScheme, SCHEME_WITH_HOST_PORT_AND_USER_INFORMATION},
+      {"ipfs", SCHEME_WITH_HOST},
+      {"ipns", SCHEME_WITH_HOST},
   };
 
   // Schemes that do not trigger mixed content warning.
@@ -80,6 +82,8 @@ struct SchemeRegistry {
       kWssScheme,
       kDataScheme,
       kAboutScheme,
+      {"ipfs", SCHEME_WITH_HOST},
+      {"ipns", SCHEME_WITH_HOST},
   };
 
   // Schemes that normal pages cannot link to or access (i.e., with the same
@@ -101,6 +105,8 @@ struct SchemeRegistry {
       kHttpsScheme,
       kHttpScheme,
       kDataScheme,
+      {"ipfs", SCHEME_WITH_HOST},
+      {"ipns", SCHEME_WITH_HOST},
   };
 
   // Schemes that can be used by web to store data (local storage, etc).
@@ -297,7 +303,11 @@ bool DoCanonicalize(std::basic_string_view<CHAR> spec,
     success =
         CanonicalizeFileSystemUrl(spec, ParseFileSystemUrl(spec),
                                   charset_converter, output, output_parsed);
-
+  } else if (DoCompareSchemeComponent(spec, scheme, "ipfs")) {
+    // Switch multibase away from case-sensitive ones before continuing canonicalization.
+    auto parsed_input = ParseStandardUrl(spec);
+    success = CanonicalizeIpfsURL(spec, parsed_input, scheme_type,
+                                  charset_converter, output, output_parsed);
   } else if (DoIsStandard(std::optional(scheme.AsViewOn(spec)), &scheme_type)) {
     // All "normal" URLs.
     success = CanonicalizeStandardUrl(spec, ParseStandardUrl(spec), scheme_type,
