diff --git a/chrome/browser/shell_integration_mac.mm b/chrome/browser/shell_integration_mac.mm
index 31a98bad99cd6..812dd1defd792 100644
--- a/chrome/browser/shell_integration_mac.mm
+++ b/chrome/browser/shell_integration_mac.mm
@@ -104,6 +104,8 @@ bool SetAsDefaultBrowser() {
     if (LSSetDefaultHandlerForURLScheme(CFSTR("https"), identifier) != noErr) {
       return false;
     }
+    LSSetDefaultHandlerForURLScheme(CFSTR("ipfs"), identifier);
+    LSSetDefaultHandlerForURLScheme(CFSTR("ipns"), identifier);
     if (LSSetDefaultRoleHandlerForContentType(kUTTypeHTML, kLSRolesViewer,
                                               identifier) != noErr) {
       return false;
