diff --git a/chrome/browser/BUILD.gn b/chrome/browser/BUILD.gn
index d090c2ac6d075..dcac3e4b06c97 100644
--- a/chrome/browser/BUILD.gn
+++ b/chrome/browser/BUILD.gn
@@ -40,6 +40,7 @@ import("//services/screen_ai/buildflags/features.gni")
 import("//services/video_effects/args.gni")
 import("//testing/libfuzzer/fuzzer_test.gni")
 import("//third_party/blink/public/public_features.gni")
+import("//third_party/ipfs_client/args.gni")
 import("//third_party/protobuf/proto_library.gni")
 import("//third_party/webrtc/webrtc.gni")
 import("//third_party/widevine/cdm/widevine.gni")
@@ -2670,6 +2671,13 @@ static_library("browser") {
       "//chrome/browser/win/installer_downloader:infobar_impl",
     ]
   }
+  if (enable_ipfs) {
+    sources += [
+      "ipfs_extra_parts.cc",
+      "ipfs_extra_parts.h",
+    ]
+    deps += [ "//components/ipfs" ]
+  }
 
   # Platforms that have a network diagnostics dialog. All others fall through
   # to the stub which is not implemented.
