diff --git a/chrome/browser/prefs/BUILD.gn b/chrome/browser/prefs/BUILD.gn
index 99b52b07ee509..c2135e18d79e6 100644
--- a/chrome/browser/prefs/BUILD.gn
+++ b/chrome/browser/prefs/BUILD.gn
@@ -7,6 +7,7 @@ import("//components/safe_browsing/buildflags.gni")
 import("//components/services/on_device_translation/buildflags/features.gni")
 import("//extensions/buildflags/buildflags.gni")
 import("//pdf/features.gni")
+import("//third_party/ipfs_client/args.gni")
 
 assert(is_win || is_mac || is_linux || is_chromeos || is_android)
 
@@ -127,7 +128,9 @@ source_set("impl") {
     "//services/preferences/public/mojom",
     "//services/preferences/tracked",
   ]
-
+  if (enable_ipfs) {
+    deps += [ "//components/ipfs" ]
+  }
   if (is_chromeos) {
     deps += [
       "//ash",
