diff --git a/content/test/BUILD.gn b/content/test/BUILD.gn
index 4d747913c7c39..d3b6b24ee7ca6 100644
--- a/content/test/BUILD.gn
+++ b/content/test/BUILD.gn
@@ -488,6 +488,7 @@ static_library("test_support") {
   configs += [
     "//build/config:precompiled_headers",
     "//v8:external_startup_data",
+    "//electron/build/config:mas_build",
   ]
 
   public_deps = [
@@ -1086,6 +1087,7 @@ static_library("browsertest_support") {
   }
 
   configs += [ "//v8:external_startup_data" ]
+  configs += ["//electron/build/config:mas_build"]
 }
 
 mojom("content_test_mojo_bindings") {
@@ -1702,6 +1704,7 @@ test("content_browsertests") {
   defines = [ "HAS_OUT_OF_PROC_TEST_RUNNER" ]
 
   configs += [ "//build/config:precompiled_headers" ]
+  configs += ["//electron/build/config:mas_build"]
 
   public_deps = [
     ":test_interfaces",
@@ -2929,6 +2932,7 @@ test("content_unittests") {
   }
 
   configs += [ "//build/config:precompiled_headers" ]
+  configs += ["//electron/build/config:mas_build"]
 
   public_deps = [ "//content:content_resources" ]
 
