diff --git a/ui/views/BUILD.gn b/ui/views/BUILD.gn
index ebc3920648cd0..43844bb1d2fb2 100644
--- a/ui/views/BUILD.gn
+++ b/ui/views/BUILD.gn
@@ -714,6 +714,7 @@ component("views") {
       "IOSurface.framework",
       "QuartzCore.framework",
     ]
+    configs += ["//electron/build/config:mas_build"]
   }
 
   if (is_win) {
@@ -1147,6 +1148,8 @@ source_set("test_support") {
     "//testing/gtest",
   ]
 
+  configs += ["//electron/build/config:mas_build"]
+
   if (is_win) {
     sources += [
       "test/desktop_window_tree_host_win_test_api.cc",
