diff --git a/url/url_canon.h b/url/url_canon.h
index 66aa4d1b1776a..39e8dc9a20918 100644
--- a/url/url_canon.h
+++ b/url/url_canon.h
@@ -840,6 +840,21 @@ bool CanonicalizeMailtoURL(const char16_t* spec,
                            CanonOutput* output,
                            Parsed* new_parsed);
 
+COMPONENT_EXPORT(URL)
+bool CanonicalizeIpfsURL(std::string_view spec,
+                         const Parsed& parsed,
+                         SchemeType scheme_type,
+                         CharsetConverter* query_converter,
+                         CanonOutput* output,
+                         Parsed* new_parsed);
+COMPONENT_EXPORT(URL)
+bool CanonicalizeIpfsURL(std::basic_string_view<char16_t> spec,
+                         const Parsed& parsed,
+                         SchemeType scheme_type,
+                         CharsetConverter* query_converter,
+                         CanonOutput* output,
+                         Parsed* new_parsed);
+
 // Part replacer --------------------------------------------------------------
 
 // Internal structure used for storing separate strings for each component.
